NAAM.md

```diff
@@ -0,0 +1,278 @@
+Alternatively, a new multicodec code can be introduced to the
+[multicodec CSV file](https://github.com/multiformats/multicodec/blob/master/table.csv) to avoid
+potential confusion between the two. The new code can then be used as the Protocol ID of Naam
+Metadata.
+
+### Updating IPNS Records
+
+IPNI offers a built-in mechanism to efficiently update the metadata associated to a list of
+multihashes without having to republish the entries. Indexer nodes uniquely identify the metadata by
+the advertisement `Provider`, and `ContextID`.
+
+Both of these values remain the same in Naam advertisements: the former is the same peer ID that
+forms part of the IPNS key, i.e. `/ipns/<peer-id>`, and the latter is fixed. As a result, to update
+an IPNS record, all that's needed is to publish a new Naam advertisement that:
+
+* links to the previous advertisement in `PrevousID` link as specified by IPNI,
+* uses the same `Provider` and `ContextID` values as before,
+* sets `IsRm` to `false`, and
+* has the updated IPNS record in Naam Metadata format.
+
+Note that there is no need to specify `Entries` when updating an existing record. It is inferred by
+the network indexers since `Provider` and `ContextID` in Naam advertisements never change.
+
+### Removing IPNS Records
+
+IPNS has EOL and TTL features built-in. This means explicit removal of records is not strictly
+required. Instead, records eventually expire. Though not strictly necessary, Naam protocol can
+utilise IPNI to explicitly remove records just like a traditional DNS nameserver.
+
+The process to remove an IPNS record published via Naam protocol is identical to removal
+advertisement specified by IPNI. Similar to a regular advertisement, an IPNS record may be removed
+by publishing an advertisement that:
+
+* links to the previous advertisement in `PrevousID` link as specified by IPNI,
+* uses the same `Provider` and `ContextID` values as before,
+* sets `IsRm` to `true`, and
+* sets `Entries` link to `NoEntries` as specified by IPNI.
+
+Once ingested, indexer nodes will eventually remove the Naam advertisement metadata along with the
+publisher information.
+
+### Resolving IPNS Records
+
+IPNS record resolution involves finding the IPNS record associated to a given IPNS key, i.e.
+`/ipns/<peer-id>`. As eluded to earlier, Naam uses the find API already provided by the network
+indexers to achieve this. To resolve an IPNS record, a resolver:
+
+* calculates the multihash of the given IPNS key as its SHA-256,
+* looks up the provider info for that multihash via the IPNI find API,
+* decodes the metadata from the provider info as IPNS record,
+* validates the record, and
+* if valid returns its value.
+
+## Limitations
+
+### IPNS Record Size
+
+The maximum size of IPNS records supported by Naam is 1 KiB, even though IPNS specification suggests
+a maximum limit of 10 KiB. The 1 KiB limit currently imposed by network indexers on the maximum size
+of advertisement metadata.
```

Comment:
Is 1KiB the  current network limit, but 100B is the recommended (you will have a good time) number? https://github.com/ipni/specs/blob/69a5f6bfdacf6050958a901324322c1e61b1611e/IPNI.md#advertisements

Reply:
The spec might be out of date though i think 100B is a reasonable value for a typical ad. the indexer nodes currently have [a hard limit of 1KiB](https://github.com/ipni/storetheindex/blob/main/api/v0/ingest/schema/schema.go#L21).